### ISSUE TITLE
Provide the validation enhancements for all Jobs without queue-name as alternative of manageJobsWithoutQueueName

### ISSUE THREAD

**Issue by tenzen-y** [kubernetes-sigs/kueue#3637]
**What would you like to be added**:
I would like to provide the validation feature for all Jobs without queue-name (`requiredJobsWithQueueName`), which rejects the Jobs without queue-name. 
However, the validations will be implemented as VAPs, and we never implement the custom validation mechanism as webhook and other APIs.

If the cluster administrator enables the `requiredJobsWithQueueName` in the Kueue Configuration API, the kueue-controller-manager will create the VAPs for all enabled Jobs.

**Why is this needed**:
The manageJobsWithoutQueueName automatically suspends all Jobs without queue-name to avoid the quota violation.
But the feature does not have good UX since it's challenging for batch users to find the reason why their Jobs are not runnable.

So, I would like to provide a way that they can easily find the reason when they submit Jobs.

**Completion requirements**:

This enhancement requires the following artifacts:

- [x] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by tenzen-y** [comment]
This is under discussion. So, any can not take this one.

**Comment by dgrove-oss** [comment]
I think LocalQueueDefaulting provides an alternate approach to solving the UX issue.  And properly dealing with child/ancestor jobs is more complex than we realized and probably beyond the scope of a VAP.  Therefore I'd suggest closing this issue as a path we are no longer planning to pursue.

**Comment by tenzen-y** [comment]
> I think LocalQueueDefaulting provides an alternate approach to solving the UX issue. And properly dealing with child/ancestor jobs is more complex than we realized and probably beyond the scope of a VAP. Therefore I'd suggest closing this issue as a path we are no longer planning to pursue.

@dgrove-oss Good point. I wanted to remove validation and enforcing mechanisms from Kueue code base to avoid violating the Kueue principle (Kueue does not implement the overwrapping features with Kube core features and famous defact OSS tools like OPA/Gatekeeper, cluster-autoscaler).

However, as you mentioned, today's Kueue manages the nested ancestors across Jobs. So, I guess that VAP can not resolve this problem. Hence, I'm closing this one for now.

/close
